src/server/game/AI/NpcBots/bot_ai.cpp

```diff
@@ -4914,7 +4914,7 @@ void bot_ai::_updateMountedState()
     if (master->IsMounted() && !me->IsMounted() && !master->IsInCombat() && !me->IsInCombat() && !me->GetVictim())
     {
         uint32 mount = 0;
-        Unit::AuraEffectList const &mounts = master->GetAuraEffectsByType(SPELL_AURA_MOUNTED);
+        Unit::AuraEffectList const &mounts = master->GetAuraEffectsByType(SPELL_AURA_MOD_INCREASE_MOUNTED_SPEED);
```

Comment:
Not all spells with `SPELL_AURA_MOD_INCREASE_MOUNTED_SPEED` are mount spells. Using items like [Carrot on a Stick](https://wotlkdb.com/?item=11122) may lead to bots not always mounting or some other inconsistent behavior.
The proper way to find mount speed would be to search through spell's effects to find `SPELL_AURA_MOD_INCREASE_MOUNTED_SPEED` or `SPELL_AURA_MOD_INCREASE_FLIGHT_SPEED` aura effect and get amount from there.